src/main/java/mate/academy/dao/BookDaoImpl.java

```diff
@@ -0,0 +1,124 @@
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(FIND_BOOK_BY_ID)) {
+            statement.setLong(1, id);
+            ResultSet resultSet = statement.executeQuery();
+            if (resultSet.next()) {
+                Book book = createBookObject(resultSet);
+                return Optional.of(book);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't find book by id " + id, e);
+        }
+        return Optional.empty();
+    }
+
+    @Override
+    public List<Book> findAll() {
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(FIND_ALL_BOOK)) {
+            List<Book> allBooks = new ArrayList<>();
+            ResultSet resultSet = statement.executeQuery();
+            while (resultSet.next()) {
+                Book bookObject = createBookObject(resultSet);
+                allBooks.add(bookObject);
+            }
+            return allBooks;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't find all books from db ", e);
+        }
+    }
+
+    @Override
+    public Book update(Book book) {
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(UPDATE_BOOK)) {
+            statement.setString(1, book.getTitle());
+            statement.setObject(2, book.getPrice());
+            statement.setObject(3, book.getId());
+            int row = statement.executeUpdate();
+            if (row < 1) {
+                throw new DataProcessingException("Expected to update row "
+                        + row + " row");
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't update book " + book, e);
+        }
+        return book;
+    }
+
+    @Override
+    public boolean deleteById(Long id) {
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement =
+                        connection.prepareStatement(DELETE_BOOK)) {
+            statement.setObject(1, id);
+            int row = statement.executeUpdate();
+            return row > 0;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't delete a book by id " + id, e);
+        }
+    }
+
+    private static Book createBookObject(ResultSet resultSet) {
+        try {
+            Book book = new Book();
+            book.setId(resultSet.getObject(ID, Long.class));
+            book.setTitle(resultSet.getString(TITLE));
+            book.setPrice(resultSet.getObject(PRICE, BigDecimal.class));
+            return book;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't create book object from DB row", e);
```

Comment:
Use informative messages for exceptions. For example, include more context when an error occurs while creating a book object from a DB row.